src/FactorAPI/ComputeEngineImpl.java

```diff
@@ -1,7 +1,50 @@
 package factorapi;
 
-public class ComputeEngineImpl implements ComputeEngine{
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class ComputeEngineImpl implements ComputeEngine {
+    private ComputeRequest request;
+
+    public ComputeEngineImpl(ComputeRequest request) {
+        this.request = request;
+    }
+
     public String executeJob(ComputeRequest request) {
```

Comment:
If I'm reading this right, this ComputeRequest request pass isn't necessary here, since the constructor requires a ComputeRequest. Not breaking obviously, but we could consider removing one of these passes